### PR TITLE
test: added integration testing for data-access

### PR DESCRIPTION
**Pull request by lafbelmonte**

### CODE REVIEW

File: koa-crud/test/integration/data-access/vendor.spec.ts [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+        const mock = {
+          name: 'Luis Angelo Belmonte',
+          type: VendorType.Transfer,
+          dateTimeCreated: new Date(),
+          dateTimeUpdated: new Date(),
+        };
+        const main = await insertOneVendor(mock);
+        await expect(selectOneVendorByFilters({ _id: main._id }))
+          .to.eventually.fulfilled.and.property('_id')
+          .eqls(main._id);
```
Comment (ralphcasipe1) [inline koa-crud/test/integration/data-access/vendor.spec.ts]: You can change this to:

```typescript
.to.eventually.fulfilled.and.property('_id', main._id);

```

File: koa-crud/test/integration/data-access/vendor.spec.ts [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+            {
+              name: 'Luis Angelo Belmonte',
+              type: VendorType.Seamless,
+              dateTimeCreated: new Date(),
+              dateTimeUpdated: new Date(),
+            },
+          ),
+        )
+          .to.eventually.fulfilled.property('type')
+          .eqls('SEAMLESS');
```
Comment (ralphcasipe1) [inline koa-crud/test/integration/data-access/vendor.spec.ts]: You can shorten this to `property('type', 'SEAMLESS')`

File: koa-crud/test/integration/data-access/vendor.spec.ts [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+      it('should throw an error', async () => {
+        const mock = {
+          name: 'Luis Angelo Belmonte',
+          type: VendorType.Transfer,
+          dateTimeCreated: new Date(),
+          dateTimeUpdated: new Date(),
+        };
+        await insertOneVendor(mock);
+        await expect(vendorExistsByFilter({ _id: 'qwe' })).to.eventually
+          .rejected;
```
Comment (ralphcasipe1) [inline koa-crud/test/integration/data-access/vendor.spec.ts]: If you want to assert the message thrown then you can use `rejectedWith('')`

**Review by ralphcasipe1 — CHANGES_REQUESTED** [review]

File: koa-crud/test/integration/data-access/vendor.spec.ts (hunk 1 above)
Comment (lafbelmonte) [reply koa-crud/test/integration/data-access/vendor.spec.ts]: Noted on this sir.

File: koa-crud/test/integration/data-access/vendor.spec.ts (hunk 2 above)
Comment (lafbelmonte) [reply koa-crud/test/integration/data-access/vendor.spec.ts]: Noted on this sir.

File: koa-crud/test/integration/data-access/vendor.spec.ts (hunk 3 above)
Comment (lafbelmonte) [reply koa-crud/test/integration/data-access/vendor.spec.ts]: Noted on this sir.

**Review by ralphcasipe1 — APPROVED** [review]
Well done
